LWM2M device level logging event configuration (#2723)
SoftwareAG/c8y-docs: content/change-logs/device-management/ui-c8y-1020-438-0-LWM2M-device-level-logging-event-configuration.md

---

> date: 2024-12-12

Reviewer: ```suggestion
date: 2024-12-19
```

title: LWM2M device level logging event configuration
product_area: Device management & connectivity
change_type:
  - value: change-QHu1GdukP
    label: Feature
component:
  - value: component-1KLUzmqfe
    label: LWM2M
build_artifact:

>   - value: tc-pjJiURv9Y
>     label: ui-c8y

Reviewer: ```suggestion
  - value: tc-ggH2M4hf3
    label: lwm2m-agent
```

ticket: DM-4190

> version: 1020.438.0

Reviewer: ```suggestion
version: 10.20.438.0
```

---

> For LWM2M devices, it is now possible to adjust logging event level. It can be adjusted based on types or completely be disabled for the device. When enabled, log events are visible in the device's events. For more details see [LWM2M configuration](/protocol-integration/lwm2m/#lwm2m-configuration). 

Reviewer: ```suggestion
For LWM2M devices, it is now possible to adjust or disable logging event level per device during the device registration or from LWM2M Configuration tab of the existing device. When enabled, log events are visible in the device's events. Lifecycle events are the default selected option. The global configuration from LWM2M Service to enabling event logging is removed.
For more details see [LWM2M device registration advanced settings](/protocol-integration/lwm2m/#advanced-settings) and [LWM2M configuration](/protocol-integration/lwm2m/#lwm2m-configuration). 
```

---
date: 2024-12-19

> title: LWM2M device level logging event configuration

Reviewer 2: ```suggestion
title: LWM2M device logging level configuration
```

product_area: Device management & connectivity
change_type:
  - value: change-QHu1GdukP
    label: Feature
component:
  - value: component-1KLUzmqfe
    label: LWM2M
build_artifact:
  - value: tc-ggH2M4hf3
    label: lwm2m-agent
ticket: DM-4190
version: 10.20.438.0
---

> For LWM2M devices, it is now possible to adjust or disable logging event level per device during the device registration or from LWM2M Configuration tab of the existing device. When enabled, log events are visible in the device's events. Lifecycle events are the default selected option. The global configuration from LWM2M Service to enabling event logging is removed.

Reviewer 2: ```suggestion
For LWM2M devices, it is now possible to disable or adjust the granularity of logging events per device during the device registration or from LWM2M Configuration tab of the existing device. When enabled, log events are visible in the device's events. Lifecycle events are the default selected option. The global configuration from LWM2M Service to enabling event logging is removed.
```

Reviewer 2: but wouldn't a customer actually prefer or expect to define the default as well? I guess most want to change the defualt to disabled and not have to go through each device.

Reviewer: @aaronraab Lifecycle events always got created. We didn't want to introduce the behavior change. Setting the tenant level defaults can be implemented later but currently we don't have such feature.

> For LWM2M devices, it is now possible to disable or adjust the granularity of logging events per device during the device registration or from LWM2M Configuration tab of the existing device. When enabled, log events are visible in the device's events. Lifecycle events are the default selected option. The global configuration from LWM2M Service to enabling event logging is removed.

Reviewer 3: ```suggestion
For LWM2M devices, it is now possible to disable or adjust the granularity of logging events per device during the device registration or from the **LWM2M configuration** tab of the existing device. When enabled, log events are visible in the device's events. Lifecycle events are the default selected option. The global configuration from the LWM2M Service for enabling event logging has been removed.
```
